vars/githubscm.groovy

```diff
@@ -125,6 +124,44 @@ def mergePR(String pullRequestLink, String credentialID='kie-ci') {
     }
 }
 
+// Optional: Pass in $BUILD_TAG as buildTag in pipeline script 
+// to trace back the build from which this tag came from.
+def tagRepository(String tagUserName, String tagUserEmail, String tagName, String buildTag = '') {
+    def currentCommit = getCommit()
+    def tagMessageEnding = buildTag ? " in build '${buildTag}'." : '.' 
+    def tagMessage = "Tagged by Jenkins${tagMessageEnding}"
+    sh("""
+        git config user.name '${tagUserName}'
+        git config user.email '${tagUserEmail}'
+        git tag -a '${tagName}' -m '${tagMessage}'
```

Comment:
In case you send a `buildTag` your message will be something like 
`'Tagged by Jenkins in build 'buildTag'.'` this double quote will be an error (I guess)
Probably
`def tagMessageEnding = buildTag ? " in build \"${buildTag}\"." : '.'` will be more appropriated

Reply:
Good catch. Fixed, thanks.